data/scripts/creaturescripts/player_death.lua

```diff
@@ -1,15 +1,33 @@
+local login = CreatureEvent("login")
+function login.onLogin(player)
+	player:registerEvent("prepareDeath")
+	return true
+end
+
+login:register()
+
+local blesses = CreatureEvent("prepareDeath")
+function blesses.onPrepareDeath(player)
+	local function blessMessage()
+		player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You were blessed with ".. player:getBlessMessage() ..".")
+	end
+	addEvent(blessMessage, 10)
+	return true
+end
+
 local playerDeath = CreatureEvent("Player Death")
 
 local deathListEnabled = true
 local maxDeathRecords = 5
 
 function playerDeath.onDeath(player, corpse, killer, mostDamageKiller, unjustified, mostDamageUnjustified)
-	local playerId = player:getId()
-	if nextUseStaminaTime[playerId] then
-		nextUseStaminaTime[playerId] = nil
+	nextUseStaminaTime[player:getId()] = 1
+
+	local function blessMessageOnDeath()
```

Comment:
Probably here you should pass the player id and check inside the function if the player exists, so we will avoid any crash.